### ISSUE TITLE
Please make Data.Tree.Forest a newtype

### ISSUE THREAD

**Issue by eugmes** [haskell/containers#440]
Making Data.Tree.Forest a newtype instead of type synonym for list would allow to create better instances for classes like Functor. Currently type of ``show <$> forest`` is ``[String]`` instead of expected ``Forest String``. One should write ``map (fmap show) forest`` to get the required effect. The same problem with things like Foldable, Traversable.

**Comment by m-renaud** [comment]
Thanks for the feature request! The pros as you mention are better instances. The downside is this would be a breaking API change so would require a major version bump.

**Comment by treeowl** [comment]
`Data.Tree` is *very old*. How old? It came around before the `containers` package existed, as part of the `base` library that shipped with GHC 6.0, which was released in May of 2003. It's not the *oldest* part of `containers`, but it's quite venerable. The proposed change would break *all non-trivial code* that uses `Data.Tree`. I say no. We are not going to do this. There is virtually nothing to be gained, and everything to be lost.

This ticket does raise a valid question: would it be less confusing to users if we stopped using the `Forest` type synonym at all? Then everyone would just see `[Tree a]` instead of `Forest a`. We could keep the synonym itself around in case anyone uses it directly.
